File: go/raindrops/raindrops.go
// Package raindrops provides the solution to the Exercism raindrops exercise.
package raindrops

import "strconv"

// Convert returns a string containing an Raindrop-ese translation of the provided number
func Convert(number int) (conversion string) {
	if number%3 == 0 {
		conversion += "Pling"
	}
	if number%5 == 0 {
		conversion += "Plang"
	}
	if number%7 == 0 {
		conversion += "Plong"
	}

	if conversion == "" {
		conversion = strconv.Itoa(number)
	}

	return
}

File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn- wordy [number factor word]
  (if (zero? (rem number factor))
    word
    ""))

(defn- plingy [number] (wordy number 3 "Pling"))
(defn- plangy [number] (wordy number 5 "Plang"))
(defn- plongy [number] (wordy number 7 "Plong"))

(defn convert [number]
  (let [converted-number
        (clojure.string/join ""
                             [(plingy number) (plangy number) (plongy number)])]
    (if (empty? converted-number)
      (str number)
      converted-number)))

